waku/v2/node/wakunode2.nim

```diff
@@ -416,6 +419,13 @@ proc mountRelay*(node: WakuNode, topics: seq[string] = newSeq[string](), rlnRela
     # TODO currently the message validator is set for the defaultTopic, this can be configurable to accept other pubsub topics as well 
     addRLNRelayValidator(node, defaultTopic)
     info "WakuRLNRelay is mounted successfully"
+  
+  if node.libp2pTransportLoops.len > 0:
```

Comment:
Hm, is this the idiomatic way of doing this? How does it work in nimbus-eth2?

Reply:
nimbus-eth2 follows a strict order where the protocol is always mounted before starting the node. In fact, they mount GossipSub when [creating the eth2 node](https://github.com/status-im/nimbus-eth2/blob/stable/beacon_chain/networking/eth2_network.nim#L1555), implying that my assumptions in the description above don't hold for them. We may consider doing the same (perhaps only for `WakuRelay`?), but that would require a fairly significant change in approach.

Seeing if related futures are set to know if the protocol/node has started doesn't feel idiomatic, but is at least not without precedent - this is similar to how [GossipSub itself avoids a "double" start](https://github.com/status-im/nim-libp2p/blob/master/libp2p/protocols/pubsub/gossipsub.nim#L583-L585). A cleaner alternative, of course, is to just add `started` state.

Reply:
In fact, I've changed this now to rather use an explicit `started` state variable (31700c8). Lmk if you have strong feelings about either the order in which we mount/start or what type of state we keep.